src/providers/map.service.ts

```diff
@@ -9,13 +9,29 @@ export class MapService {
   options = { timeout: 5000, enableHighAccuracy: true };
   markers = [];
   windows = [];
+  gmaps_key;
   constructor() {}
 
   downloadGoogleMaps(cb: Function) {
     window['mapsCb'] = cb;
     let head = document.getElementsByTagName('head')[0];
     let mapsApi = document.createElement('script');
-    mapsApi.src = `https://maps.googleapis.com/maps/api/js?libraries=places,geometry&key=${ENV.gmaps_key}&callback=mapsCb`;
+    if(this.platform.is('cordova')){
+      if(ENV === 'dev') {
+        gmaps_key = ENV.gmaps_dev;
+      } else {
+        if(this.platform.is('core') || this.platform.is('mobileweb')){
+          gmaps_key = ENV.gmaps_core;
+        }
+        else if (this.platform.is('ios')){
+          gmaps_key = ENV.gmaps_ios;
+        }
+        else {
```

Comment:
I realize that this doesn't cover all of the scenarios. If someone wants me to generate a windows API key, I will.